File: cifrado/escritorio/codigo/CSharp-WPF/CifradosGenerico/CifradosGenerico/Cifrados/ReverseG.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CifradosGenerico.Cifrados
{
    class ReverseG
    {

        public String palabra = "";
        public int grupo = 0;
        public String resultado = "";

        public int CargarRequerimientos(int valor)
        {
            return grupo = valor;
        }
        public void Cifrar()
        {
            resultado = proceso( palabra ,grupo );
        }
        public int Desifrar(String s)
        {
            return 0;
        }

        /////////////////////////////////

        public String proceso(String palabra ,int grupo)
        {
            String parte = "";
            ArrayList myAL = new ArrayList();
            if (  grupo < 2 )
            {
                return palabra;
            }
            if(grupo >= palabra.Length)
            {
                return revert(palabra);
            }
        
            foreach(int vak in Enumerable.Range(0, palabra.Length / grupo) )
            {
                String aux = palabra.Substring(vak * grupo,  grupo);
                myAL.Add(aux);
            }
        
            foreach(String vak in myAL)
            {
                parte += revert(vak);
            }
            if( palabra.Length % grupo != 0 )
            {
                return parte + " - " + palabra.Substring(parte.Length, (palabra.Length - parte.Length) );
            }
            else {
                return parte;
            }
        }

        public string revert(string s)
        {
            char[] charArray = s.ToCharArray();
            Array.Reverse(charArray);
            return new string(charArray);
        }
    }
}

File: cifrado/escritorio/codigo/Kotlin-Fx-Tornado/prueba2/src/main/kotlin/com/example/demo/app/ReverseG.kt
package com.example.demo.app

class ReverseG
{
    
    var palabra = ""
    var grupo = 0
    var resultado = ""
    fun preCifrado( grupo:Int )
    {
        this.grupo = grupo
    }
    fun cifrar()
    {
        resultado = proceso()
    }
    fun proceso():String
    {
        var parte = ""
        val list: ArrayList<String> = ArrayList()
        if(  grupo < 2 )
        {
            return palabra
        }
        if(grupo >= palabra.length)
        {
            return palabra.reversed()
        }
    
        for(vak in 0..palabra.length/grupo-1)
        {
            var aux = palabra.subSequence(vak*grupo,(vak*grupo)+grupo )
            list.add( aux.toString() )
        }
        
        for(vak in list)
        {
            parte += vak.reversed()
        }
        if( palabra.length % grupo != 0 )
        {
            return parte+ palabra.reversed().subSequence(0,palabra.length.rem(grupo) ).reversed()
        }
        else {
            return parte
        }

        
    }

}